### ISSUE TITLE
Error in splitting mesh by a plane

### ISSUE THREAD

**Issue by MNico83** [mikedh/trimesh#1248]
Hi,

I'm using trimesh to compute boolean intersection on oriented bounding box. To avoid any dependency issue, I'm trying to implement my own intersection operator by sliccing iteratively a box by supporting planes of the other box

Here, my operands
```python
m1 = trimesh.primitives.Box( extents=[10, 20, 30], transform=trimesh.transformations.random_rotation_matrix())
m2 = trimesh.primitives.Box( extents=[10, 20, 30], transform=trimesh.transformations.random_rotation_matrix())
```

then the intersection operator
```python
def intersect_box(box1 : trimesh , box2 : trimesh) -> trimesh:
    normals, _ = trimesh.triangles.normals(box1.triangles)
    current_mesh = box2
    for i,n in enumerate(normals):
        current_mesh = trimesh.intersections.slice_mesh_plane(current_mesh, -normals[i], box1.vertices[box1.faces[i][0]], cap = True)
    return current_mesh
```
However, I'm stuck by the following error : 
ValueError: Input mesh must be watertight to cap slice

Indeed, at one point , current_mesh becomes non-manifold:
![image](https://user-images.githubusercontent.com/13551294/118397460-63a1fa80-b654-11eb-9c66-cba02b77f8ef.png)

Is there a reason to this behaviour ? How can I get a way to fix my operator 
(Note : I've tried to apply all the repair methods, without success)

Thank by advance for your help!

**Comment by mikedh** [comment]
Oh yeah I guess if the meshes are simple enough you could use `slice_mesh_plane` as a slow half-plane thing, neat idea. It looks like it's trying to cap an empty section and thus failing pretty reliably (although sometimes succeeding). #1253 seems to make it through 100 of those operations pretty reliably. Also the docstring needs to be updated but you can pass multiple planes directly to `slice_mesh_plane`: 

```
import trimesh
import numpy as np


if __name__ == '__main__':

    for i in range(100):
        box1 = trimesh.primitives.Box(
            extents=[10, 20, 30],
            transform=trimesh.transformations.random_rotation_matrix())
        box2 = trimesh.primitives.Box(
            extents=[10, 20, 30],
            transform=trimesh.transformations.random_rotation_matrix())

        result = trimesh.intersections.slice_mesh_plane(
            mesh=box2,
            plane_normal=-box1.face_normals,
            plane_origin=box1.vertices[box1.faces[:, 1]],
            cap=True)
```

**Comment by mikedh** [comment]
Hey, I think this is fixed-ish by #1253, let me know if you see new errors, thanks for the report!

**Comment by MNico83** [comment]
Thanks for the fix! I've tested my code with the patch, and it works perfectly .
